### PR TITLE
romarov/accuracy gen

### PR DESCRIPTION
**Pull request by RuslanOm**
Added function ti compute accuracy on generated images

### CODE REVIEW

File: DiffAugment-stylegan2-pytorch/training/training_loop.py [hunk 1]
```diff
@@ -403,8 +404,12 @@ def training_loop(
             if rank == 0:
                 print('Evaluating metrics...')
             for metric in metrics:
+                # TODO
```
Comment (Agryosha) [inline DiffAugment-stylegan2-pytorch/training/training_loop.py]:
```suggestion
```

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ __pycache__/
 .vscode/
 *.pyc
 *.so
-*.pkl
\ No newline at end of file
+*.pkl
+.idea/
```
Comment (ActiveChooN) [inline .gitignore]:
```suggestion
.idea/

```

File: DiffAugment-stylegan2-pytorch/metrics/accuracy.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
 import dnnlib
 from . import metric_utils
 from tqdm.auto import tqdm
+from DiffAugment_pytorch import DiffAugment
+
```
Comment (ActiveChooN) [inline DiffAugment-stylegan2-pytorch/metrics/accuracy.py:15]:
```suggestion
```

File: DiffAugment-stylegan2-pytorch/training/training_loop.py [hunk 4]
```diff
@@ -209,6 +209,7 @@ def training_loop(
     if rank == 0:
         print('Setting up training phases...')
     loss = dnnlib.util.construct_class_by_name(device=device, **ddp_modules, **loss_kwargs) # subclass of training.loss.Loss
+    print(loss_kwargs)
```
Comment (RuslanOm) [inline DiffAugment-stylegan2-pytorch/training/training_loop.py]:
```suggestion
```

File: DiffAugment-stylegan2-pytorch/metrics/accuracy.py [hunk 5]
```diff
@@ -45,6 +47,27 @@ def compute_accuracy(opts, batch_size=32):
...
+def compute_accuracy_generated(opts, batch_size=32, diff_aug=False):
+    D = copy.deepcopy(opts.D).eval().requires_grad_(False).to(opts.device)
+    G = copy.deepcopy(opts.G).eval().requires_grad_(False).to(opts.device)
+
+    train_correct = 0
+    train_all = 0
+    if opts.validation_dataset_kwargs != {}:
+        all_z = torch.randn([len(opts.validation_dataset), G.z_dim], device=opts.device)
+    else:
+        all_z = torch.randn([10000, G.z_dim], device=opts.device)
```
Comment (ActiveChooN) [inline DiffAugment-stylegan2-pytorch/metrics/accuracy.py:62]: Could this magic contant be parameterized in args?

**Review by Reason239 — DISMISSED** [review]

**Review by ActiveChooN — DISMISSED** [review]

File: DiffAugment-stylegan2-pytorch/metrics/accuracy.py (hunk 5 above)
Comment (RuslanOm) [reply DiffAugment-stylegan2-pytorch/metrics/accuracy.py:62]: It's a default value

**Review by ActiveChooN — APPROVED** [review]

**Review by Reason239 — APPROVED** [review]
